tests/foreman/cli/test_syncplan.py

```diff
@@ -248,7 +248,7 @@ def test_positive_update_interval(module_org, test_data, request, target_sat):
         }
     )
     sync_plan = entities.SyncPlan(organization=module_org.id, id=new_sync_plan['id']).read()
-    request.addfinalizer(lambda: target_sat.disable_syncplan(sync_plan))
+    request.addfinalizer(lambda: target_sat.api_factory.disable_syncplan(sync_plan))
```

Comment:
@shweta83 Thanks for fixing this, we've got many other tests failing due to similar issues of helper usage(for example, I saw for `create_discovered_host`, and `update_provisioning_template`), could you check and fix all such helper usage?

Reply:
I am fixing all the issues in this PR. Moving to draft till I push all the changes in this
PR.